elyra/pipeline/component_registry.py

```diff
@@ -101,6 +111,23 @@ def get_component(self, component_id) -> Component:
             component = next(iter(component), None)
         return component
 
+    def get_categories(self) -> List[dict]:
+        category_entries: list = list()
+        with open(self._component_registry_location, 'r') as catalog_file:
```

Comment:
I feel like this should be baked into `_read_component_registry()` registry and let that return a tuple of components and categories.  I also wonder if the `_update_cache()` method would call `_read_component_registry()` directly, then perform what `get_all_xxx` was doing.  That way, we're only reading the once on each cache refresh (every 60 seconds!).  Probably not a big deal now, but when processing dozens (or hundreds) of components, it might be a good optimization.